stories/core/Table.stories.tsx

```diff
@@ -1140,17 +1147,42 @@ export const Full: Story<TableProps> = (args) => {
     [],
   );
 
+  const rowProps = useCallback(
+    (rowData: { name: string; description: string }) => {
+      return {
+        onMouseEnter: (e: React.MouseEvent) => {
+          action(`Hovered over ${rowData.name}`)(e);
+          setHoveredRow(rowData);
+        },
+        ref: (el: HTMLDivElement | null) => {
+          if (el) {
+            rowRefMap.current[rowData.name] = el;
+          }
+        },
+      };
+    },
+    [],
+  );
+
   return (
-    <Table
-      columns={columns || tableColumns}
-      data={data || tableData}
-      emptyTableContent='No data.'
-      subComponent={expandedSubComponent}
-      isRowDisabled={isRowDisabled}
-      isSelectable
-      isSortable
-      {...rest}
-    />
+    <>
+      <Table
+        columns={columns || tableColumns}
+        data={data || tableData}
+        emptyTableContent='No data.'
+        subComponent={expandedSubComponent}
+        isRowDisabled={isRowDisabled}
+        rowProps={rowProps}
+        isSelectable
+        isSortable
+        {...rest}
+      />
+      <Tooltip
+        reference={hoveredRow ? rowRefMap.current[hoveredRow.name] : undefined}
```

Comment:
This seems a bit convoluted. Tippy reference can already handle undefined, so we should be passing `ref` not `ref.current`

Reply:
I don't think I understand 😃 I still need map and `ref.current` comes from `useRef`, `ref` just gives us callback prop.

Reply:
Ah well 🤷‍♀️